tests/mcts/test_hash_canonicalization.py

```diff
@@ -0,0 +1,42 @@
+import random
+
+from rlmolecule.mcts.mcts import MCTS
+from rlmolecule.mcts.mcts_vertex import MCTSVertex
+from rlmolecule.molecule.molecule_state import MoleculeConfig
+# class MCTSHashCanonicalizationTest(unittest.TestCase):
+from tests.qed_optimization_problem import QEDOptimizationProblem
+
+
+def test_get_successors():
+    config = MoleculeConfig(max_atoms=4,
+                            min_atoms=1,
+                            tryEmbedding=False,
+                            sa_score_threshold=None,
+                            stereoisomers=False)
+    problem = QEDOptimizationProblem(config)
+    game = MCTS(problem)
+    root: MCTSVertex = game._get_root()
+    game._expand(root)
+    root.update(1.0)
+    successor0: MCTSVertex = root.children[0]
+    game._expand(successor0)
+    successor0.update(1.0)
+
+    successor1: MCTSVertex = root.children[1]
+    game._expand(successor1)
+    successor1.update(1.0)
+
+    random.seed(42)
+    for _ in range(1000):
+        game.step()
```

Comment:
`mcts.step()` got removed, it looks like it's now wrapped up in `sample()`. It might be nice to make `sample()` just do a single sample (which is an easy change), but the issue is that now we need to define and provide the `ucb_selection` function, since that's defined internally in `run` and not exposed elsewhere. 

I'm not really sure that added flexibility is really something we'll use -- we can just override `ucb_score` (as in alphazero) if we want a different selection criteria, and overriding select is easy enough in a derived class is easy enough if we need to change how selection is handled. Ultimately I think this is adding complexity where we won't really make use of it.

More specifically for this file, I refactored this test in my [test_networkx.py](https://github.com/NREL/rlmolecule/blob/psj/refactor/tests/mcts/test_networkx.py) that made it run a lot faster while still testing the core functionality. Just wondering if these changes were intentional or if this file just didn't pick up the recent changes

Reply:
I've made some changes to sample() to use a default parameter and to not require a selection function. Let me know if this resolves this issue.